ethereum/spec/src/main/java/tech/pegasys/teku/spec/schemas/SchemaRegistry.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Consensys Software Inc., 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.schemas;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.HashMap;
+import java.util.Map;
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.config.SpecConfig;
+import tech.pegasys.teku.spec.schemas.SchemaTypes.SchemaId;
+
+public class SchemaRegistry {
+  private final Map<SchemaId<?>, SchemaProvider<?>> providers = new HashMap<>();
+  private final SpecMilestone milestone;
+  private final SchemaCache cache;
+  private final SpecConfig specConfig;
+
+  public SchemaRegistry(
```

Comment:
We are relying on the builder to instantiate the registry so we can change this to package-private.